react/src/pages/MyEnvironmentPage.tsx

```diff
@@ -0,0 +1,321 @@
+import React, {
+  PropsWithChildren,
+  Suspense,
+  useState,
+  useTransition,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLazyLoadQuery, useMutation } from 'react-relay';
+
+type TabKey = 'images';
+type AdditionalImageColumnTypes = {
+  key: string;
+  imageFullName: string;
+  namespace: string;
+  lang: string;
+  baseversion: string;
+  baseimage: string;
+  constraint: string[];
+};
+
+export type CommittedImage = NonNullable<
+  MyEnvironmentPageQuery$data['customized_images']
+>[number];
+
+type CustomizedImages = CommittedImage & AdditionalImageColumnTypes;
+
+const MyEnvironmentPage: React.FC<PropsWithChildren> = ({ children }) => {
+  const { t } = useTranslation();
+  const { token } = theme.useToken();
+  const { message } = App.useApp();
+
+  const [isOpenColumnsSetting, setIsOpenColumnsSetting] = useState(false);
+  const [selectedTab] = useState<TabKey>('images');
+  const [isRefetchPending, startRefetchTransition] = useTransition();
+  const [myEnvironmentFetchKey, updateMyEnvironmentFetchKey] =
+    useUpdatableState('initial-fetch');
+  const [
+    ,
+    {
+      getNamespace,
+      getImageLang,
+      getBaseVersion,
+      getBaseImage,
+      getFilteredRequirementsTags,
+    },
+  ] = useBackendAIImageMetaData();
+
+  const { customized_images } = useLazyLoadQuery<MyEnvironmentPageQuery>(
+    graphql`
+      query MyEnvironmentPageQuery {
+        customized_images {
+          id
+          ImageNode_id
+          name
+          humanized_name
+          tag
+          registry
+          architecture
+          digest
+          labels {
+            key
+            value
+          }
+          supported_accelerators
+        }
+      }
+    `,
+    {},
+    {
+      fetchPolicy:
+        myEnvironmentFetchKey === 'initial-fetch'
+          ? 'store-and-network'
+          : 'network-only',
+      fetchKey: myEnvironmentFetchKey,
+    },
+  );
+
+  const [commitForgetAndUntag, isInflightForgetAndUntag] =
+    useMutation<MyEnvironmentPageForgetAndUntagMutation>(graphql`
+      mutation MyEnvironmentPageForgetAndUntagMutation(
+        $image_id: String!
+        $id: String!
+      ) {
+        forget_image_by_id(image_id: $image_id) {
+          ok
+          msg
+        }
+        untag_image_from_registry(id: $id) {
+          ok
+          msg
+        }
+      }
+    `);
+
+  const processedImages = _.map(customized_images, (image) => {
+    const key = image?.id;
+    const imageFullName = getImageFullName(image) || '';
+    const namespace = getNamespace(imageFullName);
+    const lang = getImageLang(imageFullName);
+    const baseversion = getBaseVersion(imageFullName);
+    const baseimage = getBaseImage(imageFullName);
+    const constraint = getFilteredRequirementsTags(imageFullName);
+
+    return {
+      ...image,
+      key,
+      imageFullName,
+      namespace,
+      lang,
+      baseversion,
+      baseimage,
+      constraint,
+    };
+  });
+
```

Comment:
In most cases in Relay, it's better to use GraphQL data without generating a new type. Transformation functions like `getImageFullName` can be called in React components or in the `render` function of columns.